server/activity/activity.controller.js

```diff
@@ -59,6 +61,17 @@ function clone({ activity, body }, res) {
   });
 }
 
+function getPreview({ activity }, res) {
+  return publishingService.fetchActivity(activity)
```

Comment:
Reference publishing helper directly instead of going through publishing service or at least implement this as static method so it does not get queued.